apps/meteor/ee/server/models/raw/LivechatUnit.ts

```diff
@@ -1,5 +1,189 @@
+		return super.deleteMany(query);
+	}
+
+	removeParentAndAncestorById(parentId: string): Promise<UpdateResult | Document> {
+		const query = {
+			parentId,
+		};
+
+		const update = {
+			$unset: { parentId: 1 },
+			$pull: { ancestors: parentId },
+		};
+
+		return this.updateMany(query, update);
+	}
+
+	async removeById(_id: string): Promise<DeleteResult> {
+		await LivechatUnitMonitors.removeByUnitId(_id);
+		await this.removeParentAndAncestorById(_id);
+
+		const query = { _id };
+		return this.deleteOne(query);
+	}
+
+	async findMonitoredDepartmentsByMonitorId(monitorId: string): Promise<FindCursor<IOmnichannelBusinessUnit>> {
+		const monitoredUnits = await this.findByMonitorId(monitorId);
+		return this.findByUnitIds(monitoredUnits, {}) as unknown as FindCursor<IOmnichannelBusinessUnit>;
+	}
+
+	async createOrUpdateUnit(
+		_id: string,
+		{ name, visibility }: { name: string; visibility: string },
+		ancestors: string[],
+		monitors: { monitorId: string; username: string }[],
+		departments: string[],
+	): Promise<IOmnichannelBusinessUnit> {
+		monitors = ([] as any[]).concat(monitors || []);
+		ancestors = ([] as string[]).concat(ancestors || []);
+
+		const record = {
+			name,
+			visibility,
+			type: 'u',
+			numMonitors: monitors.length,
+			numDepartments: departments.length,
+		};
+
+		if (_id) {
+			await this.updateOne({ _id }, { $set: record });
+		} else {
+			// @ts-expect-error someone decided that units are departments instead of creating a new collection :(
+			_id = (await this.insertOne(record)).insertedId;
+		}
+
+		ancestors.splice(0, 0, _id);
+
+		const savedMonitors = _.pluck(await LivechatUnitMonitors.findByUnitId(_id).toArray(), 'monitorId');
+		const monitorsToSave = _.pluck(monitors, 'monitorId');
+
+		// remove other monitors
+		await Promise.all(
+			_.difference(savedMonitors, monitorsToSave).map(async (monitorId) => {
+				await LivechatUnitMonitors.removeByUnitIdAndMonitorId(_id, monitorId);
+			}),
+		);
+
+		await Promise.all(
+			monitors.map(async (monitor) => {
+				await LivechatUnitMonitors.saveMonitor({
+					monitorId: monitor.monitorId,
+					unitId: _id,
+					username: monitor.username,
+				});
+			}),
+		);
+
+		const savedDepartments = _.pluck(await super.find({ parentId: _id }).toArray(), '_id');
+		const departmentsToSave = _.pluck(departments, 'departmentId');
+
+		// remove other departments
+		await Promise.all(
+			_.difference(savedDepartments, departmentsToSave).map(async (departmentId) => {
+				await super.updateOne(
+					{ _id: departmentId },
+					{
+						$set: {
+							parentId: null,
+							ancestors: null,
+						},
+					},
+				);
+			}),
+		);
+
+		await Promise.all(
+			departmentsToSave.map(async (departmentId) => {
+				await super.updateOne(
+					{ _id: departmentId },
+					{
+						$set: {
+							parentId: _id,
+							ancestors,
+						},
+					},
+				);
+			}),
+		);
+
+		return _.extend(record, { _id });
```

Comment:
let's reduce `underscore` usage 😬 
```suggestion
		return { ...record, _id };
```